crates/runtime/tests/models/search.rs

```diff
@@ -0,0 +1,320 @@
+fn normalize_search_response(mut json: Value) -> String {
+    if let Some(matches) = json.get_mut("matches").and_then(|m| m.as_array_mut()) {
+        for m in matches {
+            if let Some(obj) = m.as_object_mut() {
+                obj.remove("score");
+            }
+        }
+    }
+
+    if let Some(duration) = json.get_mut("duration_ms") {
+        *duration = json!("duration_ms_val");
+    }
+
+    sort_json_keys(&mut json);
+
+    serde_json::to_string_pretty(&json).unwrap_or_default()
+}
+
+pub(crate) fn item_tpch_dataset_w_embeddings(
+    ds_name: &str,
+    model: &str,
+    primary_keys: Option<Vec<String>>,
+    chunking: Option<EmbeddingChunkConfig>,
+) -> Dataset {
+    let mut ds_tpcds_item = get_tpcds_dataset("item", Some(ds_name), None);
+    ds_tpcds_item.embeddings = vec![ColumnEmbeddingConfig {
+        column: "i_item_desc".to_string(),
+        model: model.to_string(),
+        primary_keys,
+        chunking,
+    }];
+
+    ds_tpcds_item
+}
+
+pub(crate) fn catalog_page_tpch_dataset_w_embeddings(
+    ds_name: &str,
+    model: &str,
+    primary_keys: Option<Vec<String>>,
+    chunking: Option<EmbeddingChunkConfig>,
+) -> Dataset {
+    let mut ds_tpcds_cp = get_tpcds_dataset(
+        "catalog_page",
+        Some(ds_name),
+        Some(format!(
+            "select cp_description, cp_catalog_page_sk, cp_department, cp_catalog_number from {ds_name} limit 20"
+        ).as_str()),
+    );
+    ds_tpcds_cp.embeddings = vec![ColumnEmbeddingConfig {
+        column: "cp_description".to_string(),
+        model: model.to_string(),
+        primary_keys,
+        chunking,
+    }];
+    ds_tpcds_cp
+}
+
+pub(crate) async fn run_search(
+    app: App,
+    test_cases: Vec<SearchTestCase>,
+) -> Result<(), anyhow::Error> {
+    let _tracing = init_tracing(None);
+
+    test_request_context()
+        .scope(async {
+            let api_config = create_api_bindings_config();
+            let http_base_url = format!("http://{}", api_config.http_bind_address);
+            let rt = Arc::new(Runtime::builder().with_app(app).build().await);
+
+            let _ = init_tracing_with_task_history(None, &rt);
+
+            let rt_ref_copy = Arc::clone(&rt);
+            tokio::spawn(async move {
+                Box::pin(rt_ref_copy.start_servers(api_config, None, EndpointAuth::no_auth())).await
+            });
+
+            tokio::select! {
+                () = tokio::time::sleep(std::time::Duration::from_secs(60)) => {
+                    return Err(anyhow::anyhow!("Timed out waiting for components to load"));
+                }
+                () = Arc::clone(&rt).load_components() => {}
+            }
+
+            runtime_ready_check(&rt).await;
+            for ts in test_cases {
+                run_search_test(http_base_url.as_str(), &ts).await?;
+            }
+            Ok(())
+        })
+        .await
+}
+
+#[ignore = "Non-deterministic order of search results makes snapshot testing unreliable"]
```

Comment:
can you sort them for the test?

Reply:
The non-determinism is from the current implementation selecting one of the arbitrary `value` columns as the final value. This non-determinism cannot be solved in the tests. In the following PR, we will have a different UX for search results that can be sorted.